### PR TITLE
core: stmm: remove useless return values to local svc handlers

### PR DESCRIPTION
**Pull request by etienne-lms**
Remove the boolean return value from local functions
stmm_handle_mem_mgr_service(), stmm_handle_storage_service(),
spm_eret_error() and spm_handle_direct_req() that all end
returning to secure partition (StMM) execution.

Rename return_helper() to return_nsec_helper() and remove its
return value as it only prepare a return into normal world.

Signed-off-by: Etienne Carriere <etienne.carriere@linaro.org>

### CODE REVIEW

File: core/arch/arm/kernel/stmm_sp.c [hunk 1]
```diff
@@ -578,8 +578,8 @@ static void save_sp_ctx(struct stmm_ctx *spc, struct thread_svc_regs *svc_regs)
 }
 #endif
 
-static bool return_helper(bool panic, uint32_t panic_code,
-			  struct thread_svc_regs *svc_regs)
+static void return_nsec_helper(bool panic, uint32_t panic_code,
```
Comment (etienne-lms) [inline core/arch/arm/kernel/stmm_sp.c:581]: maybe misnamed. `return_sp_client_helper()`?  keep `return_helper()`?

Comment (jenswi-linaro) [reply core/arch/arm/kernel/stmm_sp.c:581]: If this needs to be renamed i guess `return_from_sp_helper()` would be more accurate.

File: core/arch/arm/kernel/stmm_sp.c [hunk 2]
```diff
@@ -849,17 +846,26 @@ static bool spm_handle_svc(struct thread_svc_regs *regs)
 	case FFA_VERSION:
 		DMSG("Received FFA version");
 		*a0 = MAKE_FFA_VERSION(FFA_VERSION_MAJOR, FFA_VERSION_MINOR);
-		return true;
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/stmm_sp.c:852]: Why not just return here? For this branch of this function it end here. Same for the other cases below.

**Review by jenswi-linaro — COMMENTED** [review]
When a simplification results in an increase in lines there's often reason to look a bit closer at why. In this case it's because of the dealing with `return_to_sp` in `spm_handle_svc()`.

**Comment by etienne-lms** [comment]
> When a simplification results in an increase in lines there's often reason to look a bit closer at why. In this case it's because of the dealing with return_to_sp in spm_handle_svc().

:) indeed. I wanted explicit `return_to_sp` (variable) label in the function. But maybe useless. I'll cleanup.
Thanks for the review.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
squashed + tag applied.

**Pull request merged**